woocommerce/woocommerce-gutenberg-products-block
# Update checkout route hooks to support extensions

Adds a new hook to the Checkout Route which will be used to update orders based on API request data. This is part of #4448

The new hook is named `__experimental_woocommerce_blocks_checkout_update_order_from_request` and passes on the order object, and the request data. Extensions that pass in custom request data can use this to update the order.

### How to test the changes in this Pull Request:

This new hook cannot be tested directory so just check for regressions; place an order via checkout, and test requests to the /wc/store/checkout route. Unit tests must also pass.

### Changelog

> Introduced the `__experimental_woocommerce_blocks_checkout_update_order_from_request` hook to the Checkout Store API.
